OpenSim/Actuators/VandenBogert2011Muscle.cpp

```diff
@@ -0,0 +1,1234 @@
+        projFibVel = projFibVel * optimalFiberLength;
+        projFiberLength = projFiberLength * optimalFiberLength;
+    }
+
+    double fiberVelocity = 0;
+
+    if (projFibVel != 0)
+    {
+        fiberVelocity = projFibVel * cos(projFiberLength / fiberLength);
+    }
+    else { fiberVelocity = 0; }
+
+    if (normalizedValues)
+    {
+        fiberVelocity = fiberVelocity / optimalFiberLength;
+    };
+
+    return fiberVelocity;}
+
+//------------------------------------------------------------------------------
+double VandenBogert2011Muscle::projFibVelToFiberVelocity
+        (double projFibVel, double projFibLength, bool normalizedValues=0)
+                        const {
+
+    //overload method when fiberLength is not known/provided
+
+    double fiberLength=projFibLenToFiberLength(fiberLength ,normalizedValues);
+    double fiberVelocity = projFibVelToFiberVelocity(projFibVel, fiberLength,
+                                                     projFibLength,
+                                                     normalizedValues);
+
+    return fiberVelocity;
+};
+
+
+
+//------------------------------------------------------------------------------
+VandenBogert2011Muscle::ImplicitResidual
+VandenBogert2011Muscle::calcImplicitResidual(const SimTK::State& s,
+                                             double projFibVelNorm_guess,
+                                             double activdot_guess,
+                                             double u, int returnJacobians)
+
+
+{
+    // Overload method for state as parameters
+
+    double muscleLength = getLength(s);
+
+    //double activ =getActivations(s); //Does not work because info not implemented
+    //double activ = getStateVariableValues(s,"activation");  // does not work
+    //double fiberLength = getStateVariableValues(s,"fiber_length");
+
+    SimTK::Vector stateVec =getStateVariableValues(s);
+    double fiberLength =stateVec[0];
+    double activ =stateVec[1];
+
+
+    double projFibLenNorm =
+            fiberLengthToProjectedLength(fiberLength/getOptimalFiberLength(),1);
+
+    VandenBogert2011Muscle::ImplicitResidual Results= calcImplicitResidual(
+            muscleLength, projFibLenNorm, activ, projFibVelNorm_guess,
+            activdot_guess, u, 0);
+
+    return Results;
+}
+
+
+//------------------------------------------------------------------------------
+VandenBogert2011Muscle::ImplicitResidual
+VandenBogert2011Muscle::calcImplicitResidual(SimTK::Vec2 y,
+                                             SimTK::Vec2 ydot_guess,
+                                             double muscleLength,
+                                             double u, int returnJacobians=0)
+const
+{
+    // Overload method for state vectors as parameters
+    VandenBogert2011Muscle::ImplicitResidual Results= calcImplicitResidual(
+            muscleLength, y[0], y[1], ydot_guess[0], ydot_guess[1], u,
+            returnJacobians);
+    return Results;
```

Comment:
Variable names should start with a lower case letter (except for things like transforms; `X_GB`, etc.).